server/src/test/java/org/opensearch/action/search/SearchRequestTests.java

```diff
@@ -74,6 +75,22 @@ protected SearchRequest createSearchRequest() throws IOException {
         );
     }
 
+    public void testClone() throws IOException {
+        SearchRequest searchRequest = new SearchRequest();
+        SearchRequest clonedRequest = searchRequest.clone();
+        assertEquals(searchRequest.hashCode(), clonedRequest.hashCode());
+        assertNotSame(searchRequest, clonedRequest);
+
+        SearchSourceBuilder source = new SearchSourceBuilder()
+            .fetchSource(new FetchSourceContext(true, new String[] { "field1.*" }, new String[] { "field2.*" }));
+        SearchRequest complexSearchRequest = createSearchRequest().source(source);
+        complexSearchRequest.requestCache(false);
+        complexSearchRequest.scroll(new TimeValue(1000));
+        SearchRequest clonedComplexRequest = complexSearchRequest.clone();
+        assertEquals(complexSearchRequest.hashCode(), clonedComplexRequest.hashCode());
+        assertNotSame(complexSearchRequest, clonedComplexRequest);
```

Comment:
What do you think about storing a reference to the FetchSourceContext and then updated a value inside it to see the original gets changed, but the deep copy doesn't?